### PR TITLE
Refactor typescript files dependent on node to own package

### PR DESCRIPTION
**Pull request by bryans99**
New package is sdk-rtk-node.

This means deep linking into package to get appropriate sdk is no longer required. Means that tree shaking will be supported in browser implementations.

In addition, updated looker eslint packages as lint was failing with an error. Lint now runs but there are still lots of lint errors. Will fix these in a separate PR as well as turn on full lint in CI

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -134,16 +134,12 @@
...
-      "header",
-      "prettier"
+      "@looker/eslint-config",
+      "@looker/eslint-config/license-header"
     ],
     "rules": {
-      "prettier/prettier": "error",
       "@typescript-eslint/no-explicit-any": "off",
-      "@typescript-eslint/camelcase": "off",
+      "camelcase": "off",
```
Comment (jkaster) [inline package.json:142]: That was it?!?! Well done!

File: package.json [hunk 2]
```diff
@@ -166,15 +162,17 @@
         "always",
         {
           "markers": [
-            "#"
+            "#region",
+            "#endregion"
```
Comment (jkaster) [inline package.json:166]: You're on 🔥

File: package.json [hunk 3]
```diff
@@ -166,15 +162,17 @@
...
         }
-      ]
+      ],
+      "no-use-before-define": "off"
     },
     "settings": {
       "import/resolver": {
         "typescript": {
-          "directory": "./tsconfig.json"
+          "project": "./tsconfig.json"
```
Comment (jkaster) [inline package.json:175]: Thanks for this, too. Kept meaning to get to that.

File: packages/api-explorer/scripts/utils.ts [hunk 4]
```diff
@@ -26,8 +26,8 @@
 
 import path from 'path'
 import fs from 'fs'
-import { IOauthClientApp, LookerNodeSDK } from '@looker/sdk/lib/node'
-import { NodeSettingsIniFile } from '@looker/sdk-rtl/lib/node'
+import { IOauthClientApp } from '@looker/sdk'
+import { LookerNodeSDK, NodeSettingsIniFile } from '@looker/sdk-rtl-node'
```
Comment (jkaster) [inline packages/api-explorer/scripts/utils.ts:30]: I don't know how it works to have `LookerNodeSDK` in the `sdk-rtl-node` package but I guess I'll find out further down

File: packages/sdk-rtl-node/src/nodeSdk.ts [hunk 5]
```diff
@@ -24,17 +24,9 @@
...
-  NodeSession,
-  NodeTransport,
-  IApiSettings,
-  IAuthSession,
-} from '@looker/sdk-rtl'
-import { Looker31SDK } from './3.1/methods'
-import { Looker40SDK } from './4.0/methods'
-import { environmentPrefix } from './constants'
+import { ITransport, IApiSettings, IAuthSession } from '@looker/sdk-rtl'
+import { Looker31SDK, Looker40SDK, environmentPrefix } from '@looker/sdk'
```
Comment (jkaster) [inline packages/sdk-rtl-node/src/nodeSdk.ts:28]: I think this means this package should be `@looker/sdk-node` not `@looker/sdk-rtl-node`

File: packages/sdk/src/constants.ts [hunk 6]
```diff
@@ -24,5 +24,5 @@
 
  */
 
-export const sdkVersion = '21.0'
+export const sdkVersion = '21.1'
```
Comment (jkaster) [inline packages/sdk/src/constants.ts:27]: We need to regen against 21.0 or edit to 21.0

File: packages/sdk/src/3.1/methods.ts [hunk 7]
```diff
@@ -34,9 +34,9 @@ import {
   IAuthSession,
   ITransportSettings,
   encodeParam,
-} from '@looker/sdk-rtl/lib/browser'
+} from '@looker/sdk-rtl'
 /**
- * NOTE: Do not edit this file generated by Looker SDK Codegen for Looker 21.0 API 3.1
+ * NOTE: Do not edit this file generated by Looker SDK Codegen for Looker 21.1 API 3.1
```
Comment (jkaster) [inline packages/sdk/src/3.1/methods.ts:39]: Need to regen against 21.0 or change the version stamps to 21.0

**Review by jkaster — COMMENTED** [review]
Let's discuss `@looker/sdk-node` vs. `@looker/sdk-rtl-node` as the package name

File: packages/sdk-rtl-node/src/nodeSdk.ts (hunk 5 above)
Comment (bryans99) [reply packages/sdk-rtl-node/src/nodeSdk.ts:28]: Agreed. Changing

File: packages/sdk/src/constants.ts (hunk 6 above)
Comment (bryans99) [reply packages/sdk/src/constants.ts:27]: Thanks for catching. Learning more and more about this process

File: packages/sdk-node/CHANGELOG.md [hunk 8]
```diff
@@ -0,0 +1,6 @@
+# Changelog
+
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
```
Comment (jkaster) [inline packages/sdk-node/CHANGELOG.md:6]:
```suggestion
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [Current]

- introduces `@looker/node-sdk` for using the Looker Typescript SDK with Node
```

**Review by jkaster — APPROVED** [review]
This looks much cleaner than what we had before! Nice.

You may want to take my changelog suggestion, though.

File: packages/sdk-node/src/index.ts [hunk 9]
```diff
@@ -24,10 +24,9 @@
...
  */
 
-export * from './3.1/methods'
-export * as models31 from './3.1/models'
-export * from './3.1/streams'
-export * from './4.0/methods'
-export * from './4.0/models'
-export * from './4.0/streams'
+// Version 21.0.4
+ export * from './nodeServices'
```
Comment (jkaster) [inline packages/sdk-node/src/index.ts:28]: minor nit but looks like a leading space?

**Review by jkaster — APPROVED** [review]
LGTM.

File: packages/sdk-node/src/index.ts (hunk 9 above)
Comment (bryans99) [reply packages/sdk-node/src/index.ts:28]: I'll fix that with the next PR

**Pull request merged**